File: ctutoring/assortment/book.cpp
#include <bits/stdc++.h>
using namespace std;

class Book {
private:
    string title;
    string author;
    string publisher;
    int year;
    string edition;

public:
    Book();  //default_construction
    Book(string title, string author, string publisher,
         int year, string edition);

    //mutators_change the data in private
    void settitle(const string &title);//address of string.do not change address
    void setauthor(const string &author);

    void setpublisher(const string &publisher);//
    void setyear(int year);

    void setedition(const string &edition);

    //accessors_access data in private
    string gettitle() const;

    string getauthor() const;

    string getpublisher() const;

    int getyear() const;

    string getedition() const;

};

Book::Book(){
    this->title = "";
    this->author = "";
    this->edition = "";
    this->publisher = "";
    this->year = 0;
}

Book::Book(string title, string author, string publisher,
         int year, string edition)
{
    this->title = title;
    this->author = author;
    this->publisher = publisher;
    this->year = year;
    this->edition = edition;
}

File: titlesCatalog/src/Book.java

public class Book extends Title {

    private String author;
    private String ISBN;

    public Book(){
        super();
        author="none";
        ISBN="none";
    }
    public Book(String callNumber, String title, String publisher, int year, int copies, String author, String ISBN ){
        super(callNumber,title,publisher,year,copies);
        this.author=author;
        this.ISBN=ISBN;
    }
    public String getAuthor(){
        return author;
    }
    public String getISBN(){
        return ISBN;
    }
    public String toString(){
        return super.toString() + author + "\n" + ISBN + "\n";
    }

    public void setAuthors(String author){
        this.author=author;
    }
    public void setISBN(String ISBN){
        this.ISBN=ISBN;
    }
}